authorizer-app-backend/src/main/java/org/radarbase/authorizer/resources/RestSourceUserResource.kt

```diff
@@ -0,0 +1,154 @@
+package org.radarbase.authorizer.resources
+
+import org.radarbase.authorizer.api.*
+import org.radarbase.authorizer.doa.RestSourceUserRepository
+import org.radarbase.authorizer.doa.entity.RestSourceUser
+import org.radarbase.authorizer.service.RadarProjectService
+import org.radarbase.authorizer.service.RestSourceAuthorizationService
+import org.radarbase.jersey.auth.Auth
+import org.radarbase.jersey.auth.Authenticated
+import org.radarbase.jersey.exception.HttpBadRequestException
+import org.radarbase.jersey.exception.HttpConflictException
+import org.radarbase.jersey.exception.HttpNotFoundException
+import org.radarcns.auth.authorization.Permission
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.net.URI
+import javax.annotation.Resource
+import javax.inject.Singleton
+import javax.ws.rs.*
+import javax.ws.rs.core.Context
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+
+@Path("users")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@Resource
+@Authenticated
+@Singleton
+class RestSourceUserResource(
+    @Context private val userRepository: RestSourceUserRepository,
+    @Context private val userMapper: RestSourceUserMapper,
+    @Context private val authorizationService: RestSourceAuthorizationService,
+    @Context private val projectService: RadarProjectService,
+    @Context private val auth: Auth
+) {
+
+  @GET
+  fun query(
+      @QueryParam("projectId") projectId: String?,
+      @QueryParam("sourceType") sourceType: String?,
+      @QueryParam("size") pageSize: Int?,
+      @DefaultValue("1") @QueryParam("page") pageNumber: Int): RestSourceUsers {
+
+    if (projectId != null) {
+      auth.checkPermissionOnProject(Permission.PROJECT_READ, projectId)
+    }
+
+    val queryPage = Page(pageNumber = pageNumber, pageSize = pageSize)
+    val (records, page) = userRepository.query(queryPage, projectId, sourceType)
+
+    return userMapper.fromRestSourceUsers(records, page)
```

Comment:
This should probably check for `SUBJECT_READ`. If no projectId is requested, still the individual users should be checked on whether the user has access to those. I wouldn't mind making projectId mandatory though to avoid permission escalation.
Summary:
- add `@NeedsPermission(SUBJECT, READ)` (if possible with projectId)
- add a filter for all users that are mapped: `.filter { auth.checkPermissionOnSubject(SUBJECT, READ, it.projectId, it.login)`

Reply:
I didn't make the projectId mandatory, since it would make the implementation of the REST-Sounce-Connector more complicated. At the moment, it only requests for users of a particular source-type.